apps/other/traccar.md

```diff
@@ -0,0 +1,199 @@
+networks:
+  trc2:
+    driver: bridge
+    enable_ipv6: false
+    ipam:
+      config:
+#get the network subnet
+#docker network inspect network_name      
+        - subnet: 192.168.112.0/20
+```
+
+## open root mysql shell create database, charset and priviledget user
+
+`docker exec -it traccar-db mysql -u root -p`
+
+`CREATE DATABASE IF NOT EXISTS traccar-db`
+
+`grant all privileges on traccar-db.* TO user'@'% identified by pass`
+
+`flush privileges`
+
+`ALTER DATABASE traccar CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ciALTER DATABASE traccar CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci`
+
+`\q`
+
+## reverse proxy
+
+i recoment to use caddy for reverse proxy and automatic ssl manager
+
+`nano /etc/caddy/Caddyfile`
+
+`a.random.ddns.name.duckdns.org {
+    reverse_proxy localhost:82
+}`
+
+## backup and restore traccar mysql
+
+backup
+
+`docker-compose -f  compose-file.yml exec dbname mysqldump -uroot -pYOUR_MARIADB_ROOT_PASSWORD --all-databases > dump-$(date +%F_%H-%M-%S).sql`
+
+restore
+
+`docker-compose -f compose-file.yml exec -T dbname mysql -uroot -pYOUR_MARIADB_ROOT_PASSWORD < mariadb-dump.sql`
+
+## android apps
+
+[Android client app](https://www.traccar.org/client/)
+
+On client app While service is deactivated Copy the identifier, Insert the server URL (your dens,domain), Select location accuracy (high), Select frequency report ~120 sec, disable wakelock  and enable the service
+
+On status verify that the location is updating
+
+[Android manager app](https://www.traccar.org/manager/)
+
+On server Top left menu Select the plus icon to add a device Select a random name And the identifier from the client app And save
+
+The you can select the device on the top left menu
+
+![Screenshot_20230117-214656_Traccar Manager](https://user-images.githubusercontent.com/52239579/212997717-452c1e22-b244-4f50-8806-a61daa2a9485.png)
+
+
+## Migrate google location history takeout to traccar
+
+[download your google maps location history takeout](https://takeout.google.com/takeout/custom/local_actions,location_history,maps,mymaps?)
+
+select location history only
+and json as format
+extract the Records.json
+
+you'll use a python script to limit the resaults to just time , latitude and longitude converted to proper coordinates
+
+`git clone https://github.com/Scarygami/location-history-json-converter`
+
+`cd location-history-json-converter`
+
+`pip install -r requirements.txt`
+
+`python location_history_json_converter.py Records.json  output.csv -f csv`
+
+csv output will generate Comma-separated text file with a timestamp field and a location field
+we'll add a deviceid , protocol and valid fileds to it
+
+`sed 's/^/osmand,1,/; s/$/,1/' export.csv > curated.csv`
+
+delete the "1" from the first line an replace osmand with protocol in the first line
+
+so now the file looks like
+
+```
+protocol,Time,Latitude,Longitude
+osmand,1,2012-08-25 21:26:20,37.95954620,23.72793730,1
+```
+
+5 columns
+osmand
+1
+time
+lat
+lon
```

Comment:
what's this text? should it be labelled somehow?

Reply:
It should be more well explained
Also i miss
The csv file generated by the python script contains 3 columns
time latitute longitude
After the sed command it contains 6 (and not 5) columns

@osmand,@deviceid,@Time,@Latitude,@Longitude,@valid

These represent tc_positions table fields in traccar postgres db

osand , 1 , time , lat ,lon AND 1 are the values 
Of the first row in the csv

Reply:
great, thanks for clarification. I updated the file. Have a look (if you can) and see if it still makes sense to you :-)